DEVEL.md

```diff
@@ -0,0 +1,68 @@
+```
+{"level":"info","ts":1565598764.4637527,"logger":"cmd","caller":"manager/main.go:58","msg":"Starting Istio Operator version.BuildInfo{Version:\"unknown\", GitRevision:\"unknown\", BuildStatus:\"unknown\", GitTag:\"unknown\", GoVersion:\"go1.12.5\", GoArch:\"linux/amd64\", OperatorSDK:\"v0.2.1\"}"}
+{"level":"info","ts":1565598764.4639816,"logger":"cmd","caller":"manager/main.go:62","msg":"watching all namespaces"}
+{"level":"info","ts":1565598764.4676018,"logger":"leader","caller":"leader/leader.go:55","msg":"Trying to become the leader."}
+```
+
+During development you can switch to using plain text by running the Operator with the following option:
+```bash
+--logConfig devel
+``` 
+
+This produces the following log format, which is much easier to read:
+```
+2019-08-12T10:31:20.051+0200	INFO	cmd	manager/main.go:58	Starting Istio Operator version.BuildInfo{Version:"unknown", GitRevision:"unknown", BuildStatus:"unknown", GitTag:"unknown", GoVersion:"go1.12.5", GoArch:"linux/amd64", OperatorSDK:"v0.2.1"}
+2019-08-12T10:31:20.051+0200	INFO	cmd	manager/main.go:62	watching all namespaces
+2019-08-12T10:31:20.057+0200	INFO	leader	leader/leader.go:55	Trying to become the leader.
+```
+
+## Running the operator locally
+
+During development, you can run Istio Operator locally on your dev machine and have it reconcile objects in a local or remote OpenShift cluster. You can run the Operator from within your IDE and even use a debugger.
+
+To run the Operator locally, you need to set the following environment variables:
+
+- `KUBECONFIG=<path to your kubectl config file>`
+- `POD_NAME=istio-operator`
+- `POD_NAMESPACE=istio-operator`
+- `ISTIO_CNI_IMAGE=some-registry/istio-cni-image:latest`
+- `ISTIO_CNI_IMAGE_PULL_SECRET=my-pull-secret` (required if the registry hosting the CNI image is not public; secret must be in operator's namespace as defined above)
+
+Make sure you run `tmp/build/docker_build.sh` before running the Operator to generate the Helm charts in `tmp/_output/helm`.
+
+Before running the Operator, don't forget to stop the Operator running in the cluster:
+```bash
+oc -n istio-operator scale deployment istio-operator --replicas 0
+```  
+
+Run the operator locally with the following command-line options:
+```bash
+--discoveryCacheDir /tmp/istio-operator/.kube/cache/discovery --chartPath tmp/_output/helm/istio-releases/istio-1.1.0/ istio-operator
+```
+
+### Modifying the ValidatingWebhookConfiguration 
+
+Because the Operator installs a `ValidatingWebhookConfiguration` that points to the Operator pod running inside the cluster, all CRUD operations on `ServiceMeshControlPlane` or `ServiceMeshMemberRoll` objects will fail. To fix this, you either need to remove the `ValidatingWebhookConfiguration`:
+```bash
+oc -n istio-operator delete validatingwebhookconfigurations istio-operator.servicemesh-resources.maistra.io 
```

Comment:
`-n istio-operator` shouldn't be necessary; validatingwebhookconfigurations are not namespaced

Reply:
Ah, of course. Thanks!